lib/block-supports/background.php

```diff
@@ -58,11 +58,18 @@ function gutenberg_render_background_support( $block_content, $block ) {
 	$background_styles['backgroundPosition']   = $block_attributes['style']['background']['backgroundPosition'] ?? null;
 	$background_styles['backgroundRepeat']     = $block_attributes['style']['background']['backgroundRepeat'] ?? null;
 	$background_styles['backgroundAttachment'] = $block_attributes['style']['background']['backgroundAttachment'] ?? null;
-	if ( ! empty( $background_styles['backgroundImage'] ) ) {
-		$background_styles['backgroundSize'] = $background_styles['backgroundSize'] ?? 'cover';
+	if ( isset( $background_styles['backgroundImage']['id'] ) ) {
+		$inherited_styles                  = gutenberg_get_global_styles();
+		$inherited_block_background_styles = $inherited_styles['blocks'][ $block['blockName'] ]['background'] ?? null;
+		$inherited_background_size         = $inherited_block_background_styles['backgroundSize'] ?? null;
+		if ( ! isset( $background_styles['backgroundSize'] ) && ! $inherited_background_size ) {
+			$background_styles['backgroundSize'] = 'cover';
+		}
```

Comment:
Apologies if I'm missing something, but just thinking about this one a little more. Why do we need to check the inherited styles in the server-rendered output of the block instance level styles?

In this case, the image will always be one that the user set (or that's bundled in a pattern), so in that case, won't we want to go with the defaults rather than styles inherited from global styles? I.e. if a user sets an image at the block instance level, we assume that they're setting things in the UI to their liking rather than inheriting from global styles at that point 🤔

Mostly I think my question boils down to: do we need to complexity of this look up? In the UI at least we kind of bundle setting an image with all the other properties now, so I wasn't sure how necessary it is to allow this inheritance.

Reply:
> Why do we need to check the inherited styles in the server-rendered output of the block instance level styles?

Good question. It's all very complex, you're right.

My assumption on this one is that, where the theme or global styles provide values for a block, the block should use them if no local styles exist to overwrite them.

So if a theme defines "contain" for a block but nothing else, and a user uploads a new image to that block, whether in global styles or at the block level, the default will be the same as if they chose "contain" themselves.

The other, and probably easier option, would be to just set "cover" every time a user adds an image and forget about inherited values?

Reply:
I very much appreciate the attention to detail here! I think the way you've done is the _right_ way to go about it while handling a complex situation. But my gut feeling is that it's probably more complex than we need it to be now that we have all the controls bundled together in the popover where folks can make adjustments easily.

> The other, and probably easier option, would be to just set "cover" every time a user adds an image and forget about inherited values?

That sounds like a good option to try to me!